tests/bluetooth/host/keys/bt_keys_get_addr/* fail to build

**Describe the bug**
tests/bluetooth/host/keys/bt_keys_get_addr/* fail to build in main with multiple errors

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=unit_testing ../tests/bluetooth/host/keys/bt_keys_get_addr
3. ninja
4. See error

**Expected behavior**
No build errors, tests passing

**Impact**
Failing tests

**Logs and console output**
```

[1/11] Building C object CMakeFiles/testbinary.dir/src/test_suite_full_list_no_overwrite.c.obj
In file included from tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_no_overwrite.c:9:
subsys/bluetooth/host/keys.h:224:51: warning: ‘struct bt_bond_info’ declared inside parameter list will not be visible outside of this definition or declaration
  224 | void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info, void *user_data),
      |                                                   ^~~~~~~~~~~~
[2/11] Building C object mocks/CMakeFiles/mocks.dir/id.c.obj
In file included from tests/bluetooth/host/keys/mocks/../mocks/id.h:10,
                 from tests/bluetooth/host/keys/mocks/id.c:8:
subsys/bluetooth/host/keys.h:224:51: warning: ‘struct bt_bond_info’ declared inside parameter list will not be visible outside of this definition or declaration
  224 | void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info, void *user_data),
      |                                                   ^~~~~~~~~~~~
[3/11] Building C object mocks/CMakeFiles/mocks.dir/keys_help_utils.c.obj
In file included from tests/bluetooth/host/keys/mocks/keys_help_utils.c:9:
subsys/bluetooth/host/keys.h:224:51: warning: ‘struct bt_bond_info’ declared inside parameter list will not be visible outside of this definition or declaration
  224 | void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info, void *user_data),
      |                                                   ^~~~~~~~~~~~
[4/11] Building C object CMakeFiles/testbinary.dir/src/test_suite_full_list_invalid_values.c.obj
FAILED: CMakeFiles/testbinary.dir/src/test_suite_full_list_invalid_values.c.obj 
/usr/bin/gcc -DZTEST_UNITTEST -Isubsys/testsuite/ztest/include/zephyr -Isubsys/testsuite/ztest/unittest/include -Isubsys/testsuite/include/zephyr -Isubsys/testsuite/ztest/include -Isubsys/testsuite/include -Iinclude/zephyr -Iinclude -I. -Itests/bluetooth/host/keys/mocks/.. -Itests/bluetooth/host -Isubsys/bluetooth -Isubsys/bluetooth/host -m32 -imacros build/zephyr/include/generated/zephyr/autoconf.h -Wall -I build/zephyr/include/generated -g -gdwarf-4 -include ztest.h -MD -MT CMakeFiles/testbinary.dir/src/test_suite_full_list_invalid_values.c.obj -MF CMakeFiles/testbinary.dir/src/test_suite_full_list_invalid_values.c.obj.d -o CMakeFiles/testbinary.dir/src/test_suite_full_list_invalid_values.c.obj -c tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c
In file included from tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c:9:
subsys/bluetooth/host/keys.h:224:51: warning: ‘struct bt_bond_info’ declared inside parameter list will not be visible outside of this definition or declaration
  224 | void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info, void *user_data),
      |                                                   ^~~~~~~~~~~~
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c: In function ‘bt_keys_find_key_in_use_invalid_cases_test_find_key_in_use_receives_null_conn_ref’:
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c:78:42: error: assignment to ‘void (*)(enum bt_conn_type,  void (*)(struct bt_conn *, void *), void *)’ from incompatible pointer type ‘void (*)(int,  void (*)(struct bt_conn *, void *), void *)’ [-Wincompatible-pointer-types]
   78 |         bt_conn_foreach_fake.custom_fake = bt_conn_foreach_conn_ref_null_custom_fake;
      |                                          ^
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c: In function ‘bt_keys_find_key_in_use_invalid_cases_test_find_key_in_use_receives_null_data_ref’:
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c:107:42: error: assignment to ‘void (*)(enum bt_conn_type,  void (*)(struct bt_conn *, void *), void *)’ from incompatible pointer type ‘void (*)(int,  void (*)(struct bt_conn *, void *), void *)’ [-Wincompatible-pointer-types]
  107 |         bt_conn_foreach_fake.custom_fake = bt_conn_foreach_data_ref_null_custom_fake;
      |                                          ^
[7/11] Building C object CMakeFiles/testbinary.dir/src/main.c.obj
In file included from tests/bluetooth/host/keys/bt_keys_get_addr/src/main.c:9:
subsys/bluetooth/host/keys.h:224:51: warning: ‘struct bt_bond_info’ declared inside parameter list will not be visible outside of this definition or declaration
  224 | void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info, void *user_data),
      |                                                   ^~~~~~~~~~~~
[8/11] Building C object CMakeFiles/testbinary.dirsubsys/bluetooth/host/keys.c.obj
In file included from subsys/bluetooth/host/hci_core.h:10,
                 from subsys/bluetooth/host/keys.c:28:
include/zephyr/devicetree.h:3564:9: warning: "DT_NODE_HAS_STATUS" redefined
 3564 | #define DT_NODE_HAS_STATUS(node_id, status) \
      |         ^~~~~~~~~~~~~~~~~~
In file included from include/zephyr/app_memory/app_memdomain.h:9,
                 from subsys/testsuite/ztest/include/zephyr/ztest_test.h:16,
                 from subsys/testsuite/ztest/include/zephyr/ztest.h:44,
                 from <command-line>:
include/zephyr/linker/linker-defs.h:34:9: note: this is the location of the previous definition
   34 | #define DT_NODE_HAS_STATUS(node, status) 0
      |         ^~~~~~~~~~~~~~~~~~
include/zephyr/devicetree.h:3587:9: warning: "DT_NODE_HAS_STATUS_OKAY" redefined
 3587 | #define DT_NODE_HAS_STATUS_OKAY(node_id) DT_NODE_HAS_STATUS(node_id, okay)
      |         ^~~~~~~~~~~~~~~~~~~~~~~
include/zephyr/linker/linker-defs.h:35:9: note: this is the location of the previous definition
   35 | #define DT_NODE_HAS_STATUS_OKAY(node) 0
      |         ^~~~~~~~~~~~~~~~~~~~~~~
[9/11] Building C object CMakeFiles/testbinary.dir/src/test_suite_full_list_overwrite_oldest.c.obj
FAILED: CMakeFiles/testbinary.dir/src/test_suite_full_list_overwrite_oldest.c.obj 
/usr/bin/gcc -DZTEST_UNITTEST -Isubsys/testsuite/ztest/include/zephyr -Isubsys/testsuite/ztest/unittest/include -Isubsys/testsuite/include/zephyr -Isubsys/testsuite/ztest/include -Isubsys/testsuite/include -Iinclude/zephyr -Iinclude -I. -Itests/bluetooth/host/keys/mocks/.. -Itests/bluetooth/host -Isubsys/bluetooth -Isubsys/bluetooth/host -m32 -imacros build/zephyr/include/generated/zephyr/autoconf.h -Wall -I build/zephyr/include/generated -g -gdwarf-4 -include ztest.h -MD -MT CMakeFiles/testbinary.dir/src/test_suite_full_list_overwrite_oldest.c.obj -MF CMakeFiles/testbinary.dir/src/test_suite_full_list_overwrite_oldest.c.obj.d -o CMakeFiles/testbinary.dir/src/test_suite_full_list_overwrite_oldest.c.obj -c tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c
In file included from tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c:9:
subsys/bluetooth/host/keys.h:224:51: warning: ‘struct bt_bond_info’ declared inside parameter list will not be visible outside of this definition or declaration
  224 | void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info, void *user_data),
      |                                                   ^~~~~~~~~~~~
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c: In function ‘bt_keys_get_addr_full_list_overwrite_oldest_test_full_list_all_keys_in_use’:
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c:138:42: error: assignment to ‘void (*)(enum bt_conn_type,  void (*)(struct bt_conn *, void *), void *)’ from incompatible pointer type ‘void (*)(int,  void (*)(struct bt_conn *, void *), void *)’ [-Wincompatible-pointer-types]
  138 |         bt_conn_foreach_fake.custom_fake = bt_conn_foreach_all_keys_in_use_custom_fake;
      |                                          ^
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c: In function ‘bt_keys_get_addr_full_list_overwrite_oldest_test_full_list_no_keys_in_use’:
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c:168:42: error: assignment to ‘void (*)(enum bt_conn_type,  void (*)(struct bt_conn *, void *), void *)’ from incompatible pointer type ‘void (*)(int,  void (*)(struct bt_conn *, void *), void *)’ [-Wincompatible-pointer-types]
  168 |         bt_conn_foreach_fake.custom_fake = bt_conn_foreach_no_keys_in_use_custom_fake;
      |                                          ^
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c: In function ‘bt_keys_get_addr_full_list_overwrite_oldest_test_full_list_key_0_in_use_key_1_oldest’:
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c:202:42: error: assignment to ‘void (*)(enum bt_conn_type,  void (*)(struct bt_conn *, void *), void *)’ from incompatible pointer type ‘void (*)(int,  void (*)(struct bt_conn *, void *), void *)’ [-Wincompatible-pointer-types]
  202 |         bt_conn_foreach_fake.custom_fake = bt_conn_foreach_key_slot_0_in_use_custom_fake;
      |                                          ^
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c: In function ‘bt_keys_get_addr_full_list_overwrite_oldest_test_full_list_key_0_in_use_key_2_oldest’:
tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c:244:42: error: assignment to ‘void (*)(enum bt_conn_type,  void (*)(struct bt_conn *, void *), void *)’ from incompatible pointer type ‘void (*)(int,  void (*)(struct bt_conn *, void *), void *)’ [-Wincompatible-pointer-types]
  244 |         bt_conn_foreach_fake.custom_fake = bt_conn_foreach_key_slot_0_in_use_custom_fake;
      |                                          ^
ninja: build stopped: subcommand failed.
```


**Environment (please complete the following information):**

 - OS: Linux Ubuntu 24.04
 - Toolchain: host gcc 14.2
 - Commit SHA or Version used: main as of now 1689eaf6a4551165905979772d981ed6713acef5

**Additional context**
Part of the build failure is due to a regression introduced in 82d8f09de18708259b85e5bbf5e32541fd807ed4

--- a/subsys/bluetooth/host/keys.h
+++ b/subsys/bluetooth/host/keys.h
@@ -6,6 +6,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef ZEPHYR_SUBSYS_BLUETOOTH_HOST_KEYS_H_
+#define ZEPHYR_SUBSYS_BLUETOOTH_HOST_KEYS_H_
+
+#include <zephyr/bluetooth/bluetooth.h>
+
 /** @cond INTERNAL_HIDDEN */
 
 enum bt_keys_type {
@@ -233,3 +238,5 @@ void bt_keys_link_key_update_usage(const bt_addr_t *addr);
 void bt_keys_show_sniffer_info(struct bt_keys *keys, void *data);
 
 /** @endcond */
+
+#endif /* ZEPHYR_SUBSYS_BLUETOOTH_HOST_KEYS_H_ */

--- a/tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c
+++ b/tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c
@@ -28,14 +28,14 @@ static int bt_unpair_unreachable_custom_fake(uint8_t id, const bt_addr_le_t *add
 	return 0;
 }
 
-static void bt_conn_foreach_conn_ref_null_custom_fake(int type, bt_conn_foreach_cb func,
-							void *data)
+static void bt_conn_foreach_conn_ref_null_custom_fake(enum bt_conn_type type,
+						      bt_conn_foreach_cb func, void *data)
 {
 	func(NULL, data);
 }
 
-static void bt_conn_foreach_data_ref_null_custom_fake(int type, bt_conn_foreach_cb func,
-							void *data)
+static void bt_conn_foreach_data_ref_null_custom_fake(enum bt_conn_type type,
+						      bt_conn_foreach_cb func, void *data)
 {
 	struct bt_conn conn;
 

--- a/tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c
+++ b/tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c
@@ -46,8 +46,8 @@ static int bt_unpair_unreachable_custom_fake(uint8_t id, const bt_addr_le_t *add
 	return 0;
 }
 
-static void bt_conn_foreach_key_slot_0_in_use_custom_fake(int type, bt_conn_foreach_cb func,
-							  void *data)
+static void bt_conn_foreach_key_slot_0_in_use_custom_fake(enum bt_conn_type type,
+							  bt_conn_foreach_cb func, void *data)
 {
 	struct bt_conn conn;
 
@@ -71,8 +71,8 @@ static void bt_conn_foreach_key_slot_0_in_use_custom_fake(int type, bt_conn_fore
 	func(&conn, data);
 }
 
-static void bt_conn_foreach_all_keys_in_use_custom_fake(int type, bt_conn_foreach_cb func,
-							void *data)
+static void bt_conn_foreach_all_keys_in_use_custom_fake(enum bt_conn_type type,
+							bt_conn_foreach_cb func, void *data)
 {
 	struct bt_conn conn;
 
@@ -86,8 +86,8 @@ static void bt_conn_foreach_all_keys_in_use_custom_fake(int type, bt_conn_foreac
 	}
 }
 
-static void bt_conn_foreach_no_keys_in_use_custom_fake(int type, bt_conn_foreach_cb func,
-							void *data)
+static void bt_conn_foreach_no_keys_in_use_custom_fake(enum bt_conn_type type,
+						       bt_conn_foreach_cb func, void *data)
 {
 	struct bt_conn conn;
 

